### DISCUSSION TITLE
Search useability

### DISCUSSION THREAD

**Discussion by eu-genia** [SChAth/ScAthiop#1597] [Ideas]
Following up on the discussion - some ideas to make the various Search options make more accessible to the larger audience

- [ ] Have a separate menu point (can stay under Resources - or top level, but again the width may be an issue) for Search, where all Searches have sub-menu-points
Menu: SEARCH
Submenu 1: Simple search
Submenu 2: Facet search
Submenu 3: Advanced search
Submenu 4: X-Path query
Submenu 5: SPARQL query

- [ ]  Introduce in the search pages the definitions currently hidden under general Help (https://betamasaheft.eu/help.html) - i.e. 
Reproduce the definitions on the specific search pages, e.g. the page https://betamasaheft.eu/xpath should have on top at least the text as present now
XPath search
All searches are XPath searches in an XML database like the one behind this website, but what you can ask in a XPath query is much more than any of the provided search interfaces can offer. So, if you know how to write your XPath, and know the source TEI (available for each file, by appending .xml to the identifier of the record) you will then be able to see your results listed.
Here you can avail yourself also of the structure of the file, the relative position of elements, and the nested elements, querying across the axes.
Examples:
person record which have at least some attribute for birth and death (can be when, notBefore, notAfter) elements and occupation type ruler'.
Manuscripts with an addition element typed Ownership Note followed by another one with type Supplication.
- [ ] Provide more ready-to-go search examples of various types and various degree of granularity where the users can "simply" subsitute variables
(I am not sure whether it is possible to make the SPARQL search examples more "editable", or one can only do it in the address line such as "https://betamasaheft.eu/sparql?query=SELECT+%3Fms+%3Fperson%0D%0AWHERE+%7B%0D%0A++%3Fannotation+a+bm%3Adonor+%3B%0D%0A++++++++++++++oa%3AhasBody+%3Fperson+%3B%0D%0A++%09%09%09++oa%3AhasTarget+%3Fms+.%0D%0A++%3Fms+a+bm%3Amss+.%0D%0A++%3Fperson+foaf%3Agender+%27female%27+.%0D%0A%7D+")

**Comment by PietroLiuzzo** [comment]
thanks, that is very useful.
please @SophiaD-M also add here your ideas about how to make searches easier.

**Comment by smaugustine** [comment]
Regarding issues with the "less-advanced" searches, especially relating to the indexing and ranking of results, is it worth considering an alternative search engine to eXistDB's own functions? I'm thinking of options like [ElasticSearch](https://www.elastic.co/) or [Algolia](https://www.algolia.com/), professional search tools that have more sophisticated methods of indexing and ranking data. Perhaps our data is subject to frequent updates and reindexing, but I'm not sure it's worth inventing our own solutions when there are existing options with years of development and refinement behind them. Just an idea.

**Comment by smaugustine** [comment]
Feature request for XPath search: every time you run an XPath search, the input field is cleared. It would be nice if the query remained in the input field so you could modify it or adjust it more easily (rather than having to retype it).

**Comment by PietroLiuzzo** [comment]
@smaugustine your idea is to change DB and indexing... well, BM will need to change developer for that (@abausi), I am definitely not going to develop newly the entire research environment just to use fancier technical solutions and waste time in development which could be invested otherwise for more functionality, more data. Also exist-db has years of development and refinement, as well as a flexibility in data handling hardly comparable to any other resource which is the reason we used it to begin with, and we are not alone, besides being tightly related to the TEI community. I also think it would not bring any actual benefit over the Lucene indexes we already have, compared to a process of continuous improvement and optimization which we carry out on a daily basis. You are of course free to develop your own application using those tools over the BM data in any of the provided formats. also, please make a new issue for a new issue.

**Comment by smaugustine** [comment]
@PietroLiuzzo To be clear, I wasn't my intention to suggest so drastic a change. Database and search are two separate things. Just as an example to clarify, we see that these are separate offerings within single companies: Amazon DynamoDB and Amazon ElasticSearch, MongoDB and Atlas Search. I would never suggest that we use anything other than eXistDB for our database, and I think we have a great database, honestly. I was simply inquiring about options for searching. Beta Masaheft already has a data API, feeding the data into a new index wouldn't require so much change. If it's impractical and better that we continue to work with our existing indexes, okay. It was just an open suggestion.

**Comment by PietroLiuzzo** [comment]
we clearly have very different Weltanschauungen. I simply do not think Elastic Search or similar would help solve this issue. if you can practically demonstrate it would in some respect, I am happy to consider that and go on with any change which will bring measurable practical improvement. I will also ask on the exist db channels if there are others using elastic search in addition to these indexes to gather more intelligence. The indexes defined in exist can be improved in number and quality, the queries on them as well. feeding data to a new index, changing queries to look at that index where appropriate, maintain additional indexes in the flow so they are in sync with the others, etc. _are_ much of a change IMHO.

**Comment by SophiaD-M** [comment]
Dear all,

I apologize for not using the correct technical terms, as I do not know/understand them. I simply think, very plainly phrased, that BM is a search engine, with an amazing backbone of information. But if this information is not accessible it is not worth much.* The accessibility should be the focus here, and while it would be wonderful for all users to get acquainted with the IT behind BM, we cannot expect the to have time or interest in this. The average user has probably very different interests. He uses BM to find either a catalogue, or a certain manuscript, an art theme, or wants to compare _incipit_ of Miracles or so. 

Are there ways to see what people most frequently search for? I would take this as a start and create a quick search, as Zhenia suggested. 
I imagine a set of search options which pre-set search parameters for the given topic, in which the user only needs to fill in the keyword he is looking for. (See my amateurish image) 
<img width="957" alt="BM search" src="https://user-images.githubusercontent.com/29302681/101461097-b9060d00-393a-11eb-848a-3a5dbe2b3853.png">

If they do not find what they are looking for they still have the advanced search and can explore on their own. But I think this would ease the access to the most important feature of the database that people are interested in.

My understanding of BM might be wrong, but I thought it is a tool for people to use, said usage should be easy and as barrier-free as possible, otherwise they will stop at one point. The main advantage of BM is making information available to the wider public that was otherwise contained in books that have a much more limited availability.** Think about all Ethiopian colleagues - how many catalogues are accessible in Ethiopian libraries? Few! To this (and other issues) BM is the remedy.





*do you know this joke? "Where do you best hide a dead body? On the second page of Google search results."
** Kropp 1988, "Armenische Osterfestberechnung... Russica non leguntur!" is another perfect illustration to my point.

**Comment by abausi** [comment]
I was CC by Sophia; I understand that this is again the output of the yesterday session (that I could not attend), which confirms that it was very useful. I support the idea that the resources must be as clearly understandable and as simply accessible as possible. Out of experience in every field, those who develop a system are not in the best situation to understand what is generally needed and which are the challenges, and in the end it is good that the focus of the developer remains on the quality of the data. This is a complex issue. In the end, however, the search for a more simply way of making resources accessible is extremely important and all voices claiming that more has to be done towards this direction must be heard.

**Comment by PietroLiuzzo** [comment]
I have done some experiment using only the currently available search field on the home page, so limited to that functionality.

TL;DR summary: I am really struggling to understand why a one simple access point would be more complex than seven entries as in the example in the picture. 

1. if I search a shelfmark https://betamasaheft.eu/facet.html?query=%22BML+Acq.+e+doni+38%22 I get one and the correct result. The shelf mark entered can also be slightly incorrect if one uses the manuscript list https://betamasaheft.eu/manuscripts/list?mainname=BML+Acq.+e+doni+38 

2. if I search a common name of work e.g. miracles of mary
     *  https://betamasaheft.eu/facet.html?query=%22Miracles+of+Mary%22 I get more results, but that is why there are indeed more things which respond to the parameter.
     *  I can search also with a transliterated name written in a non formal transliteration e.g. Taamera Maryam https://betamasaheft.eu/facet.html?query=%22Taammera+Maryam%22 and I do get many results to filter in, among which several relevant ones.

3. if I search for an art theme, in order to find an illumination, e.g. Covenant of Mercy, https://betamasaheft.eu/facet.html?query=%22Covenant+of+Mercy%22 the authority file is my first hit. the page tells me what keywords it is related to and where exactely. I can click show attestations to see places where this is used outside of the miniature context. The relation between art theme and keyword is stated right at the top. Here we could do with some more links to additional views, e.g. like this https://betamasaheft.eu/authority-files/list?keyword=AT1036MercyCovenant and https://betamasaheft.eu/decorations?legendText=&otherText=&target-artTheme=AT1036MercyCovenant which organise the same information slightly differently.

4. mixing the search in the dictionary with the search in the text database I think would be wrong, plus, each and every single geez word provides a direct link to a search for it. and if one does not want to click there, a double click will popup the results without even going to the Dillmann...

5. for places and persons, I have tried searching for Addis Ababa and Sarsa Dengel notice I have omitted the "" for this experiment
     * https://betamasaheft.eu/facet.html?query=Addis+Ababa  there are two immediately relevant results. Here the presence of contributed material from IslHornAfr may be confusing, but we are speaking of first and third result in the list... you have in the result view, as always direct links to list of manuscripts etc. the record, once clicked gives you maps, a list of churches as potential repositories of manuscripts, a link to the places index where you can see the use of that place name. https://betamasaheft.eu/IndexPlaces?pointer=LOC1216AddisA
     * https://betamasaheft.eu/facet.html?query=Sarsa+Dengel needs a bit more filtering assuming I am looking for the king. I filtered only by occupation type https://betamasaheft.eu/facet.html?query=Sarsa+Dengel&occupation-facet=ruler and https://betamasaheft.eu/persons/PRS8550sardaDe/main is my first result, which links to several informations including bibliography directly available, the person index for the attestations of the name, wikidata and viaf (!) the graph view and analytic view of that record offer additional insights, for example role attestation and all we know of relations with other persons displayed ad table and graph.

**Comment by PietroLiuzzo** [comment]
> * Have a separate menu point (can stay under Resources - or top level, but again the width may be an issue) for Search, where all Searches have sub-menu-points

could this be please unpacked? I do not understnd

**Comment by PietroLiuzzo** [comment]
> * Introduce there the definitions currently hidden under general Help (https://betamasaheft.eu/help.html)

there where? in the menu? 

please remember that last year, after a lot of discussion we opted for diminishing the load on a page, we are now going full speed in the opposite direction by loading with text and information parts of the app, like the navigation bar, which are always available. can we instead consider expanding the "explore" module?

**Comment by PietroLiuzzo** [comment]
as for the sparql endpoint editable, it is possible, not easy. this is the best example I know https://query.wikidata.org/ 
but I find it much harder at the end of the day than simply going for the query.

**Comment by eu-genia** [comment]
> > * Introduce there the definitions currently hidden under general Help (https://betamasaheft.eu/help.html)

Introduce on each search dedicated page. 

Would that really slow down the process so much if https://betamasaheft.eu/xpath where we have the text 
"Enter here your XPath 3.0 query to the data. Please, use t: namespace for TEI elements. The starting point of any Xpath should be $config:collection-root if you are searching the entire dataset.
"
see three more lines of text on top?

**Comment by PietroLiuzzo** [comment]
ok, now I understand what you mean, not in the menu, in the page with the search.

**Comment by eu-genia** [comment]
> > * Have a separate menu point (can stay under Resources - or top level, but again the width may be an issue) for Search, where all Searches have sub-menu-points

Menu: SEARCH
Submenu 1: Simple search
Submenu 2: Facet search
Submenu 3: Advanced search
Submenu 4: X-Path query
Submenu 5: SPARQL query

**Comment by eu-genia** [comment]
> * if I search for an art theme, in order to find an illumination, e.g. Covenant of Mercy, https://betamasaheft.eu/facet.html?query=%22Covenant+of+Mercy%22 the authority file is my first hit. the page tells me what keywords it is related to and where exactely. I can click show attestations to see places where this is used outside of the miniature context. The relation between art theme and keyword is stated right at the top. Here we could do with some more links to additional views, e.g. like this https://betamasaheft.eu/authority-files/list?keyword=AT1036MercyCovenant and https://betamasaheft.eu/decorations?legendText=&otherText=&target-artTheme=AT1036MercyCovenant which organise the same information slightly differently.
> *

I think this is a faulty example. Try looking for Miracle of Mary as art item instead. For a person who DOES NOT KNOW ANYTHING about keywords. Just landed on BM.
I understand what Sophia means - and this is something we talked about - that one can search directly in one type of records.
The limitations of faceted search are that you must search for something first to restrict it, but one can get around there by starting from sample searches maybe. Or having the search for art item button bring one directly to https://betamasaheft.eu/decorations

**Comment by eu-genia** [comment]
In general, there are those who think everyone who comes on BM has to prepare themselves, first study the guidelines, learn how the data is organized, understand XML/Xquery/Xpath/RDF, and then start searching/looking for information, and those who think that everyone who comes on BM should be able to more or less intuitively get around without first learning additional skills.

We already had the story of ANNIS that is there for the TraCES data but that noone is ever going to use, because of the necessity to learn, and remember, the query language first. Let us try to find a compromise.

(It is not by chance that most Ethio-SPaRe project members still look for E-S MSS on DomLib with a traditionally organized single- or multi-parameter search)

**Comment by PietroLiuzzo** [comment]
- the list searches have the same parameter approach as EthioSpare, based on the same data, what is the difference? 
- what else would you like which you miss from DomLib functionality? BTW: if it was missing anyone of those preferring the old website could have asked it at any time in the past five years. it is never too late!
- the advanced search offers most of what you are asking, including searching without starting from a string. people complained about it. we offer the facet search as well (not instead of!) and still people are generically unhappy. 
- what happens which is so disconcerting if you are searching for "miracles of mary" as art theme? this I do not understand, please give me one concrete example. Especially, how do you think that the application should know, from your string only that you have in mind an art theme? As far as I know It simply cannot. In the reply it will tell you about all it can tell you and then you have to do some filtering as in any other search engine. If you know of search engines that can read users minds I am happy to learn... 
- Searches directly for the types of records and entities and subpart are also available. let us make them more visible, as already suggested and more accessible, change their style, their functionality, but I need you to tell me what you want. 

I personally do not think one has to understand anything before getting to the website. you get to the website, you search, you read (?). Users have all available information. it is a lot of information, so you need to get your head around it. But it is also a research environment, not a clothes  shop. We also provide all sorts of support for that engagement. 
It goes without saying that if you do know what you are looking for you can find things quicker and easier. You need to know what the app offers, you need to try, in some cases you need to read some documentation. Is this different from what is required by any other resource online? Room for improvement is always large, but I do not think we can avoid these steps and pretend we can find a solution which will satisfy everyone and do magic.

**Comment by SophiaD-M** [comment]
Thank you Zhenia, yes I also see those two extremes.
I admit Pietro that your examples above did not convince me. 
I made another test, for an article I was searching for an image of St. Yared. I started by simply searchign for "Yared" on the home site, and then tried to adjust the parameters to find such an image. It took me 3 minutes (I stopped the time) to find a ms that does contain said image. At least I think, for some reason the images of the mss are not loading: https://betamasaheft.eu/manuscripts/EMIP00497/main#viiv(erso)

I would have probably needed only 1 minute to find several images on Mazgaba Seelat, and also DomLib. In DomLib, I admit, I would have also used the advanced search, and not the simple search. But the advanced is still much easier to navigate, simply because it has way less options than BM.

**Comment by eu-genia** [comment]
> * what happens which is so disconcerting if you are searching for "miracles of mary" as art theme? this I do not understand, please give me one concrete example. Especially, how do you think that the application should know, from your string only that you have in mind an art theme? As far as I know It simply cannot. In the reply it will tell you about all it can tell you and then you have to do some filtering as in any other search engine. If you know of search engines that can read users minds I am happy to learn...

There is too much misunderstanding, I am sorry.
I was following the idea of Sophia: the users coming from the outside do not know what is hidden where. They do not know anything about types of records, keywords, or what and how they can search for. Of course we can say that they should spend some time reading the guidelines, the help page, studying the structure behind, and then start their queries. Or we can guide them, showing them from the very beginning, directly, what and how can be looked for, at least basically. 
When I come on BM I have no idea that "miracle of mary" as art theme exists and can be searched for. But if I see the search for art theme button, or field, where, if I entry a word, I only look inside this type of record/authority file, then I will know.

**Comment by smaugustine** [comment]
I wonder that it's possible to throw together a graphical interface that would build SPARQL queries without the user having to worry so much about the technical aspect? Using buttons, input fields, and select boxes that correspond to elements of common SPARQL queries? SQL and other database interfaces often have such query builders.

So instead of having to type `SELECT * WHERE { bm: PRS10350Yeshaq  bm:isAuthorOfEthiopicTranslation ?object } LIMIT 5` you could have something like:

<img width="609" alt="Screen Shot 2020-12-08 at 13 09 50" src="https://user-images.githubusercontent.com/6315671/101482217-a994bd00-3956-11eb-8714-bc5931d37aa1.png">

**Comment by PietroLiuzzo** [comment]
> I wonder that it's possible to throw together a graphical interface that would build SPARQL queries without the user having to worry so much about the technical aspect? Using buttons, input fields, and select boxes that correspond to elements of common SPARQL queries? SQL and other database interfaces often have such query builders.

please read above. 

this works only for extremely simple sparql queries. similar results can be obtained in several other ways already in the application

**Comment by eu-genia** [comment]
> * what else would you like which you miss from DomLib functionality? BTW: if it was missing anyone of those preferring the old website could have asked it at any time in the past five years. it is never too late!

What I meant is e.g. this:
![image](https://user-images.githubusercontent.com/17987728/101482870-a1894d00-3957-11eb-9be8-c2cea86d81ea.png)

Without knowing anything I can look for all manuscripts on a specific subject with a specific word in Title, type of palaeography mentioned, a scribe I know, of a specific size range, with e.g. finding aids present, etc. etc. - all this without knowing XPath

**Comment by smaugustine** [comment]
@PietroLiuzzo 
> this works only for extremely simple sparql queries. similar results can be obtained in several other ways already in the application

Fair, in the simplest of cases. But I would still suggest that having buttons corresponding to common keywords used in queries would be advantageous. The very editor we're using to type these comments is an example of this: we could type `###` for heading but still there's a button on top of the editor to do that for us.

See, for example, the popular phpmyadmin interface for MySQL: https://images.app.goo.gl/Cb2NY1oWZav2V3qb7.

If people aren't doing SPARQL queries often, but occasionally do need to, I think it helps not to require them to remember everything. Sure there's documentation, but again there's documentation for GitHub markdown and still they have a gui for it (again, simple example, but consider how many query builders exist for various database platforms).

**Comment by PietroLiuzzo** [comment]
> When I come on BM I have no idea that "miracle of mary" as art theme exists and can be searched for. But if I see the search for art theme button, or field, where, if I entry a word, I only look inside this type of record/authority file, then I will know.

and why is this better/simpler than just searching "miracle of mary" and finding out the application has a lot to say about that?

I do not think we are in the realm of "misunderstanding", as I am trying quite hard to pose questions which will make me understand and provide feedback to act upon. But if you are trying to be gratuituossly aggressive towards me and criticise without grounds my work, you are managing well.

The screenshot you share is not different from this
<img width="1063" alt="Schermata 2020-12-08 alle 13 21 30" src="https://user-images.githubusercontent.com/508012/101483369-5facd680-3958-11eb-833d-0861143c38f5.png">
or this
<img width="1680" alt="Schermata 2020-12-08 alle 13 21 59" src="https://user-images.githubusercontent.com/508012/101483385-650a2100-3958-11eb-878d-78b69ccb32b9.png">

if there are things missing, you can point them and they can be added.

**Comment by PietroLiuzzo** [comment]
> @PietroLiuzzo
> 
> > this works only for extremely simple sparql queries. similar results can be obtained in several other ways already in the application
> 
> Fair, in the simplest of cases. But I would still suggest that having buttons corresponding to common keywords used in queries would be advantageous. The very editor we're using to type these comments is an example of this: we could type `###` for heading but still there's a button on top of the editor to do that for us.

the filter by relation name in the facet search and the search in relations do just the same for encoded relations. but see below.

> 
> See, for example, the popular phpmyadmin interface for MySQL: https://images.app.goo.gl/Cb2NY1oWZav2V3qb7.
> 
> If people aren't doing SPARQL queries often, but occasionally do need to, I think it helps not to require them to remember everything. Sure there's documentation, but again there's documentation for GitHub markdown and still they have a gui for it (again, simple example, but consider how many query builders exist for various database platforms).

most of the Sparql queries that in time became evidently required are embedded in the views, the roles, the relations, etc. so, the sparql endpoint is simply not used, to be honest, no matter how much I advertise it. There is no generic "need" to use it in most cases IMHO, as the information is the same which is present in the XML. If this becomes a clear requirement, I will build that. then let us see if it will be used by anyone or it will just produce one more thing to be unhappy about.

**Comment by eu-genia** [comment]
> and why is this better/simpler than just searching "miracle of mary" and finding out the application has a lot to say about that?

because I may never get to the page in search results with any miracle of mary as an art theme from the general search. you know, we know, that it is encoded and can be searched for, but a John Doe from the outside does not and will possibly never find out.

> The screenshot you share is not different from this

I would disagree. The BM search is not at all the same from the GUI point of view.  
I am not saying it is better. But it is more transparent for an outside user.

**Comment by eu-genia** [comment]
I wonder if more people should be asked to contribute to the issue with their input or it would make it even more chaotic :-)

**Comment by PietroLiuzzo** [comment]
> because I may never get to the page in search results with any miracle of mary as an art theme from the general search. 

and your proposed solution is a dedicated search on the home page which says "art themes"? and what would you expect that to search in? which kind of results would you expect? only hits in the art themes? manuscripts containing that art theme?


> I would disagree. The BM search is not at all the same from the GUI point of view.
> I am not saying it is better. But it is more transparent for an outside user.

please point out how.

**Comment by PietroLiuzzo** [comment]
> I wonder if more people should be asked to contribute to the issue with their input or it would make it even more chaotic :-)

his conversation is public as any other here, anyone can contribute.

**Comment by smaugustine** [comment]
> But if you are trying to be gratuituossly aggressive towards me and criticise without grounds my work, you are managing well.

I'm just going to step back and say that I am not being aggressive or critical in any way, nor would it ever be my intention.

People have asked about ways to help simplify the more advanced types of searches and I'm simply trying to present ideas to bridge things between the very technically advanced side of things and average everyday people just trying to find the data they're looking for.

So far, I only have presented very open suggestions, "What about?", "Would it be helpful if?". If it's a dumb idea or something we already have or something not worth implementing, then that's a show of my own ignorance. I don't see how it becomes some sort criticism or attack.

**Comment by eu-genia** [comment]
so please @SusanneHummel @antobrita @DariaElagina @DenisNosnitsin1970 @MarcinKrawczuk @Ralph-Lee-UK if you have some ideas as to how the access to search options can be improved (if it should be improved - maybe it is perfect as it is) - use this issue to share, thank you!

**Comment by PietroLiuzzo** [comment]
sorry @smaugustine the entire conversation, not your suggestions, which are more constructive than others indeed in many respects, is addressed to me instead of being a common reflection. I did not intend that as a line for you specifically, I am sorry if it came through like this, it was more in reply of what @eu-genia wrote.
I will take the same position I have taken in other similar cases. I will just not participate in this conversation for a while until people have made their requirements clear, then I will do my slave job and realise what is realistically possible.

**Comment by eu-genia** [comment]
> But if you are trying to be gratuituossly aggressive towards me and criticise without grounds my work, you are managing well.

I was not criticizing at all. I am awfully sorry if anything came across as a criticism.
I just tried to reconcile the two points of view trying to see if there is a way in-between and trying to find examples to highlight the difficulties for external users Sophia was trying to highlight - but it now sounds as if there is none.

I will step back as well, AFAIK we can leave all as it is.

**Comment by PietroLiuzzo** [comment]
> Thank you Zhenia, yes I also see those two extremes.
> I admit Pietro that your examples above did not convince me.
> I made another test, for an article I was searching for an image of St. Yared. I started by simply searchign for "Yared" on the home site, and then tried to adjust the parameters to find such an image. It took me 3 minutes (I stopped the time) to find a ms that does contain said image. At least I think, for some reason the images of the mss are not loading: https://betamasaheft.eu/manuscripts/EMIP00497/main#viiv(erso)
> 
> I would have probably needed only 1 minute to find several images on Mazgaba Seelat, and also DomLib. In DomLib, I admit, I would have also used the advanced search, and not the simple search. But the advanced is still much easier to navigate, simply because it has way less options than BM.

only because I had already started this reply.
BM does not provide any sort of search for or in images as far a I know. you can reach images from descriptions. we can look at how to implement that if it becomes a precise requirement for an image search of some kind with specific features.

**Comment by PietroLiuzzo** [comment]
please @eu-genia it would be useful for the clarity of this issue not to leave the many questions opened by your points un-answered. if you could spend few minute replying to each I am sure many more precise requirements would come up so that they can be listed, split into separate issue and carried out.

**Comment by eu-genia** [comment]
I have updated the top message with my first input after the meeting, for the rest I am at a loss as I have the impression nothing is possible or sensible, so leave it to the others.

While we are at it - I was also wondering - I know again we have here something we designed "together" - but I now have the feeling that part of this is because some new users may not even understand that the main page is scrollable down. The arrow and the menu button are visible for me but do many use them?

@SophiaD-M have you ever scrolled down to e.g. https://betamasaheft.eu/#furtherSearchOptions? Should there be a way to make the homepage content which is already there more visible?

**Comment by PietroLiuzzo** [comment]
@eu-genia please reply to the questions about the DomLib search above.

**Comment by SophiaD-M** [comment]
I did not in fact! That might be because I am using my laptop only? Look at this screenshot, the arrow pointing down is not even visible on my screen.

![image](https://user-images.githubusercontent.com/29302681/101490367-8839ce00-3962-11eb-80d1-0b12d7b47679.png)

**Comment by abausi** [comment]
It was obviously impossible for me to follow the discussion. I think 
this is one of the cases when a discussion in person (on Zoom) is the 
onyl way to progress. I remain with the impression that more mutual 
understanding is needed and that there are real and genuine needs that 
must be considered.

Il 08.12.2020 14:32, Pietro Liuzzo ha scritto:

**Comment by eu-genia** [comment]
> @eu-genia please reply to the questions about the DomLib search above.

I am sorry I probably cannot explain well what I meant.
Of course on BM you can find the same things. What I meant you need a bit more time and patience to get to the result. And more luck if you do not know what you are looking for.
Again - **I am not at all saying it is any better**. BM is better as it offers much more. But obviously it has its limitations. 

In a "traditional" very plain search of DomLib you do search from the very beginning in only one type of entity, you see which fields exist, you can search in any, or all, or any combination, with the GUI hinting as to whether you are looking for an exact or a partial match, all those things, in one click. 
(E.g. for the search example in DomLib above, looking for MSS with apocrypha, with Mary in title, a particular term in palaeographic part, a name element of a scribe, finding aid, width less than 100 - I only fill the form and get 5 MSS as a result, immediately, 
![image](https://user-images.githubusercontent.com/17987728/101492574-43fbfd00-3965-11eb-90da-96655efdc407.png)

In BM I have to go through a lot of filters (which would not give me the same results, as I have no option of free text search in separate fields) or do an XPath query which would but is not so straightforward for outsiders

I understand that BM has much more complex data structure and has much more to offer so such "simplified" solutions are not possible but maybe there are ways to make it useable also to non-specialists. E.g. having pre-set sample searches for different types of entities that one can then modify could possibly help. But maybe others have better ideas as to what can make the VRE more transparent.
